### PR TITLE
add annotation support to aws-ebs-csi pods

### PR DESCRIPTION
**Pull request by avish42**

### CODE REVIEW

**Comment by ashish1099** [comment]
This will fix #13600

File: upup/models/cloudup/resources/addons/aws-ebs-csi-driver.addons.k8s.io/k8s-1.17.yaml.template
Comment (ashish1099) [inline upup/models/cloudup/resources/addons/aws-ebs-csi-driver.addons.k8s.io/k8s-1.17.yaml.template:398]: this should be `range` https://github.com/kubernetes/kops/pull/13536/files#diff-387221074177e2de294fd64375019fcfcae1519aebd3b97d2c9dc7ae9426a53fR131

Comment (avish42) [reply upup/models/cloudup/resources/addons/aws-ebs-csi-driver.addons.k8s.io/k8s-1.17.yaml.template:398]: ok

**Comment by avish42** [comment]
@k8s-ci-robot /retest

**Comment by KlavsKlavsen** [comment]
@olemarkus Can you approve above checks to run? (it says first time contributors needs approval to run checks - we need this support in kops - to solve #13541 :)

**Comment by olemarkus** [comment]
This only add annotations to the controller, but not the daemonset. Maybe you want to distinguish between the two?

**Comment by KlavsKlavsen** [comment]
@olemarkus Good catch. @avish42 will add it, so one can also set same annotation for daemonset. the AWS Helm chart for this, does exactly that - sets same annotation for both controllers and nodes - so IMHO the same is needed here - they both need to use the same role in AWS.

**Comment by avish42** [comment]
@olemarkus can you please approve it now - if it's all okay

File: pkg/apis/kops/v1alpha3/componentconfig.go
Comment (olemarkus) [inline pkg/apis/kops/v1alpha3/componentconfig.go:890]:
```suggestion
	// PodAnnotations are the annotations added to AWS EBS CSI node and controller Pods.
```

**Review by olemarkus — CHANGES_REQUESTED** [review]
Just want to make it more explicit that the annotation is added to _all_ pods. Then we are good, I think.

**Review by olemarkus — APPROVED** [review]

**Comment by KlavsKlavsen** [comment]
@olemarkus those tests that fail - fail with:
```
 ok  	k8s.io/kops/util/pkg/vfs/tests	0.037s
FAIL
make: *** [Makefile:170: test] Error 1
Error: Process completed with exit code 2.
```

Not much detail about whats actually failing in this test? anyone who can enlighten us here :)

**Comment by hakman** [comment]
@avish42 Would you mind squashing and rebasing the PR? Thanks!

**Comment by hakman** [comment]
Until tests pass, to avoid noise, canceling /lgtm.
/lgtm cancel

**Comment by avish42** [comment]
@hakman looks like unit tests are passing

**Comment by hakman** [comment]
Thanks @avish42. Please also squash the commits.

**Pull request merged**
